homeassistant/components/rainbird/config_flow.py

```diff
@@ -67,6 +74,40 @@ def async_get_options_flow(
         """Define the config flow to handle options."""
         return RainBirdOptionsFlowHandler()
 
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauthentication upon an API authentication error."""
+        self.host = entry_data[CONF_HOST]
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm reauthentication dialog."""
+        errors: dict[str, str] = {}
+        if user_input:
+            try:
+                _LOGGER.debug(
+                    "Reauthenticating to Rain Bird controller: %s", user_input
+                )
+                serial_number, wifi_params = await self._test_connection(
+                    self.host, user_input[CONF_PASSWORD]
+                )
+            except ConfigFlowError as err:
+                _LOGGER.error("Error during config flow: %s", err)
+                errors["base"] = err.error_code
```

Comment:
Appears there is not full test coverage on config flow which is mandatory.

Reply:
Thanks, added more test coverage.